component/src/main/java/org/wso2/extension/siddhi/io/jms/sink/JMSPublisher.java

```diff
@@ -19,74 +19,76 @@
 package org.wso2.extension.siddhi.io.jms.sink;
 
 import org.apache.log4j.Logger;
-import org.wso2.carbon.messaging.CarbonMessage;
-import org.wso2.carbon.messaging.MapCarbonMessage;
-import org.wso2.carbon.messaging.TextCarbonMessage;
-import org.wso2.carbon.messaging.exceptions.ClientConnectorException;
-import org.wso2.carbon.transport.jms.sender.JMSClientConnector;
+import org.wso2.carbon.transport.jms.contract.JMSClientConnector;
+import org.wso2.carbon.transport.jms.exception.JMSConnectorException;
 import org.wso2.carbon.transport.jms.utils.JMSConstants;
 
 import java.io.UnsupportedEncodingException;
 import java.nio.ByteBuffer;
 import java.util.HashMap;
 import java.util.Map;
+import javax.jms.BytesMessage;
+import javax.jms.JMSException;
+import javax.jms.MapMessage;
+import javax.jms.Message;
+import javax.jms.TextMessage;
 
 /**
  * JMS publisher which creates the message and sends to JMS.
  */
 public class JMSPublisher implements Runnable {
     private static final Logger log = Logger.getLogger(JMSPublisher.class);
-
-    // this field has to be set for the TextCarbonMessage for it to work
-    private static final String MESSAGE_TYPE_FIELD = "JMS_MESSAGE_TYPE";
-    private static final String TEXT_MESSAGE_TYPE = "TextMessage";
-    private static final String BYTE_ARRAY_MESSAGE_TYPE = "ByteMessage";
-
     private Map<String, String> jmsProperties;
     private JMSClientConnector jmsClientConnector;
-    private CarbonMessage message;
+    private Message message;
 
     public JMSPublisher(String destination, Map<String, String> staticJMSProperties,
-            JMSClientConnector jmsClientConnector, Object payload) throws UnsupportedEncodingException {
+                        JMSClientConnector jmsClientConnector, Object payload)
+            throws UnsupportedEncodingException {
         this.jmsProperties = new HashMap<>();
         this.jmsProperties.putAll(staticJMSProperties);
         this.jmsProperties.put(JMSConstants.PARAM_DESTINATION_NAME, destination);
         this.jmsClientConnector = jmsClientConnector;
-        this.message = handleCarbonMessage(payload);
+        try {
+            this.message = handleMessage(payload);
+        } catch (JMSConnectorException | JMSException e) {
+            log.error("Failed to process payload: " + e.getMessage());
```

Comment:
Log the exception as well, along with the message to see the stack trace.

Reply:
Fixed in commit [#3fb99a6822b8f5bb32d97ebdc4b7495aa01a0dfc](https://github.com/wso2-extensions/siddhi-io-jms/commit/3fb99a6822b8f5bb32d97ebdc4b7495aa01a0dfc)